client/mysqldump.c

```diff
@@ -570,6 +572,16 @@ static struct my_option my_long_options[] =
   {"secure-auth", OPT_SECURE_AUTH, "Refuse client connecting to server if it"
     " uses old (pre-4.1.1) protocol. Deprecated. Always TRUE",
     &opt_secure_auth, &opt_secure_auth, 0, GET_BOOL, NO_ARG, 1, 0, 0, 0, 0, 0},
+  {"start-sql-file", OPT_START_SQL_FILE, "Execute SQL statements from the file"
```

Comment:
It is not a problem to do this, but this will lead to the series of #ifdefs which are code smells. Current implementation is generic, so it says nothing about Debug Sync. It is just executing sql file on start and on the end. How to use it in production? I don't know. i.e. you can drop all tables after doing dump ;-). I would say that adding it permanently does not cause problems. Additionally it allows us to test mysqldump in release builds as well. 
@percona-ysorokin what is your opinion?

Reply:
Not sure if this method is 100% suitable here but please consider
https://dev.mysql.com/doc/refman/5.7/en/server-system-variables.html#sysvar_init_connect
I think you can create a non-privileged user for `mysqldump` for thich this init script will be executed whereas for `root` connections (default one in MTR) it will be not.

Reply:
OK, I have removed these options from release build as initially suggested by @dutow .
The idea wiht sysvar_init_connect would work in this test case if not the one small problem:
1. For users that have the SUPER privilege, the content of init_connect is not executed.
2. Setting DEBUG_SYNC value requires the SUPER privilege.
So when connecting as non-SUPER, server starts execution of SET DEBUG_SYNC=... but fails because of privileges check.

Reply:
Yeah, you are right. I haven't thought about this.